### DISCUSSION TITLE
"[E053] Could not read config.cfg from /content/drive/MyDrive/project/SpaCy_model/config.cfg"

### DISCUSSION THREAD

**Discussion by Ykisi** [explosion/spaCy#8596] [Help: Coding & Implementations]
![image](https://user-images.githubusercontent.com/77232538/124395390-883d5900-dd0c-11eb-9391-5f3672812854.png)

Hi!

I train a model (Using Colab) and saved it. Now I try to load it and I get this message:
"[E053] Could not read config.cfg from /content/drive/MyDrive/project/SpaCy_model/config.cfg"

I've attached a screenshot of the saved model hierarchy (which is the output of "nlp.to_disk(output_dir)")

Why do I get this message?  I just want to load the saved model :(

**_my training code_**

nlp=spacy.load("en_core_web_sm")
ner=nlp.get_pipe('ner')

ner.add_label('tdate')

optimizer = nlp.entity.create_optimizer()
other_pipes = [pipe for pipe in nlp.pipe_names if pipe != 'ner']
with nlp.disable_pipes(*other_pipes) :
  for itn in range(40):
    # shuffle examples before training
    random.shuffle(TRAIN_DATA)
    # batch up the examples using spaCy's minibatch
    batches = minibatch(TRAIN_DATA, size=compounding(4., 32., 1.001))
    # dictionary to store losses
    losses = {}
    for batch in batches:
      texts, annotations = zip(*batch)
      # Calling update() over the iteration
      nlp.update(texts, annotations, sgd=optimizer, drop=0.35, losses=losses)
    print("Losses", losses)

**Comment by svlandeg** [comment]
Hi!

What is the version of spaCy you ran to train the model? And are you using the same version when loading it back in?

I'm asking because it looks like you've saved the model with spaCy v2, but are trying to read it back in with spaCy v3, and unfortunately the two aren't compatible.

**Reply by Ykisi** [reply]
Hi!

Yah, I thought that might be the problem. 
weird because I loaded the model 3 days after saving it, guess the version changed in those 3 days.

I tried to train again with v3 version (code attached below) but it performs worse then my v2 saved model. any guess why?

Also, I understand that you recommend working with the train configuration, but I work with Colab notebook - can it be done there? (I don't know how to work with your train cfg file in colab, is it possible?)

Thanks for the help!

**My V3 training code:** (the same as V2 with adjustments, and it performs worse!)

nlp=spacy.load("en_core_web_sm")
ner=nlp.get_pipe('ner')

ner.add_label('tdate')

optimizer = nlp.create_optimizer()
other_pipes = [pipe for pipe in nlp.pipe_names if pipe != 'ner']
with nlp.disable_pipes(*other_pipes) :     
  for itn in range(50):
    # shuffle examples before training
    random.shuffle(TRAIN_DATA)
    # batch up the examples using spaCy's minibatch
    batches = minibatch(TRAIN_DATA, size=compounding(4., 32., 1.001))
    # ictionary to store losses
    losses = {}
    for batch in batches:
        for text, annotations in batch:
            # create Example
            doc = nlp.make_doc(text)
            example = Example.from_dict(doc, annotations)
            # Update the model
            nlp.update([example], losses=losses, drop=0.5)
    print("Losses", losses)

**Reply by polm** [reply]
> Also, I understand that you recommend working with the train configuration, but I work with Colab notebook - can it be done there? (I don't know how to work with your train cfg file in colab, is it possible?)

Colab allows you to save local files, so you can just create the config locally to use it. Like any notebook, you can preface a line of code with `!` to run it as a shell command rather than as Python code.

**Reply by svlandeg** [reply]
> I tried to train again with v3 version (code attached below) but it performs worse then my v2 saved model.

Can you give some more details? How big is your training set, how big is your evaluation set, and what do the performance numbers in both cases actually look like?
